### PR TITLE
Added content about web browsers

### PR DESCRIPTION
**Pull request by bharmalh**
I added a new topic called "What is a web browser" and added content and resources to it. Will really appreciate any feedback.

### CODE REVIEW

File: content/topics/what-is-a-web-browser/index.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: What is a web browser?
+keywords: web browsers, chromium, chrome, safari, firefox, browsers explained
+description: A collection of informational pieces that explains what web browsers are and their different types.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:4]:
```suggestion
description: An informational page that explains what web browsers are and the difference between the types of browsers on the market.
```

File: content/topics/what-is-a-web-browser/index.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+menu:
+  main:
+    parent: topics
+    name: What is a web browser?
+---
+
+A web browser, in simple terms, is an app on your computer, phone, tablet or other internet-enabled devices that allows you to get data from the internet and displays it to you.
+
+For example, when you type in 'www.google.com' and hit enter, the browser gets the necessary files (HTML, CSS, Javascript, Images and more) from Google's servers and paints up the page (rendering) which is shown to you.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:13]: Let's. "promote" one of the more privacy-centric websites out there :)

```suggestion
For example, when you type in 'www.duckduckgo.com' and hit enter, the browser gets the necessary files (HTML, CSS, Javascript, Images, and more) from DuckDuckGo's servers, combines, parses, and renders (or paints) the page with which you can then interact with.
```

File: content/topics/what-is-a-web-browser/index.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This article breaks down the different parts and features of a browser with simple explanations and images: [Internet Basics - Using a Web Browser](https://edu.gcfglobal.org/en/internetbasics/using-a-web-browser/1/ "Internet Basics - Using a Web Browser")
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:15]: I think the links need to be like this.

```suggestion
This article breaks down the different parts and features of a browser with simple explanations and images: [Internet Basics - Using a Web Browser](https://edu.gcfglobal.org/en/internetbasics/using-a-web-browser/1/)
```

It would also be neat to embed the video directly on our page; what do you think?

```
{{< youtube id="FxirRVJWUTs" title="A video introduction to browser basics" >}}
```

Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:15]: Also, I think we can just link to the page. Something like:

```suggestion
* Internet Basics - Using a Web Browser](https://edu.gcfglobal.org/en/internetbasics/using-a-web-browser/1/ "Internet Basics - Using a Web Browser")
```

File: content/topics/what-is-a-web-browser/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Chromium-based browsers
+
+Chromium is a free and open-source browser, which means that the code used to create it is freely available and written by contributors. Then, different groups/people can play with it and shape it to whatever form they like, in the process creating their own browser.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:19]:
```suggestion
Chromium is a free and open-source browser, which means that the source code used to create it is freely available and written by contributors. Because of its open-source nature, anyone can contribute to the Chromium project or, create a fork (copy) of the project to create their own version of the browser.
```

Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:19]: It would also be good to link to the Chromium project page here.

File: content/topics/what-is-a-web-browser/index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+The most popular example of a chromium-based browser is [Google Chrome](https://www.google.com/chrome/ "Google Chrome"). However, a lot of other browsers today are built on top of Chromium, including Brave, Microsoft Edge, Vivaldi and Samsung Internet.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:21]: I believe we should also use a capital letter when referring to Chromium.

Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:21]: Would also be nice to link to those various browsers.

File: content/topics/what-is-a-web-browser/index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+One major advantage of using Chromium-based browsers is that extensions that work with Google Chrome work with them too!
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:23]: Web extensions have become a standard, and as such extensions generally now work across all browsers. With that in mind, I feel that we should remove this sentence.

File: content/topics/what-is-a-web-browser/index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+To understand more about chromium you can read this [article by LifeWire](https://www.lifewire.com/chromium-web-browser-4171288 "article byLifeWire").
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:25]:
```suggestion
### Additional reading

* [Chromium source on GitHub](https://github.com/chromium/chromium)
* [Chromium explained on Wikipedia](https://en.wikipedia.org/wiki/Chromium_(web_browser))
* [Deep dive video on the difference between Chrome and Chromium](https://www.youtube.com/watch?v=xxuioJkc0J8)
```

File: content/topics/what-is-a-web-browser/index.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+## WebKit-based browsers
+
+WebKit is the browser engine used mainly by the Safari Browser as well as many other apps on macOS, iOS and Linux. A browser engine is a core part of every major browser responsible for converting HTML and other resources into what you see on your screen.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:29]:
```suggestion
WebKit is the browser engine used mainly by Safari. Webkit is maintained by Apple computers and mainly used by the Safari web browser. A browser engine is a core part of every major browser responsible for converting HTML and other resources into what you see on your screen.
```

File: content/topics/what-is-a-web-browser/index.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+Safari is the major browser on all Apple devices including iPhones, iPads and Macs. You might also be surprised to know that, right now, all browsers on iOS use Safari as their rendering engine. This means that even if you install Chrome on iOS, it is still Safari. The only thing that is different is the UI and UX provided by Chrome for example. For web developers, if you find a bug in Safari on iOS, it will also exist in Chrome on iOS.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:31]: I believe that when referring to Safari on mobile devices, one should always state Safari on iOS.

Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:31]: We could also mention that Webkit is open source.

File: content/topics/what-is-a-web-browser/index.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+You can read more about WebKit [here](https://webkit.org/ "here") and on its Wikipedia [page](https://en.wikipedia.org/wiki/WebKit "page").
+
+## Mozilla Firefox
+
+Mozilla Firefox is a free and open-source browser developed by the Mozilla Foundation. At its core is the Gecko browser engine similar to WebKit for Safari as mentioned above.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:37]: The last part of this sentence might cause a bit of confusion. I would rephrase it like this:

```suggestion
Mozilla Firefox is a free and open-source browser developed by the Mozilla Foundation. As mentioned, Safari browsers use the Webkit rendering engine. What we did not mention is that Chromium-based browsers use a fork of Webkit called Blink. At the core of the Firefox browser is the Gecko rendering engine as well as component of the [Servo](https://servo.org/) engine.
```

File: content/topics/what-is-a-web-browser/index.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+Firefox brands itself as a secure and privacy-focused web browser that doesn't sell its user's data to advertisers.
+
+This [FAQ](https://www.mozilla.org/en-US/firefox/faq/) from Mozilla provides answers to some common questions about the browser.
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:41]: I would combine these into a single paragraph.

File: content/topics/what-is-a-web-browser/index.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+Mozilla's history is quite interesting as it is seen as a 'spiritual successor' to Netscape Navigator, a historic and popular browser of the early internet. The documentary below is a really interesting look into the story behind the creation of Firefox.
+
+{{< youtube 4Q7FTjhvZ7Y >}}
```
Comment (schalkneethling) [inline content/topics/what-is-a-web-browser/index.md:45]: We should always use the following form of the Youtube shortcode for [accessibility reasons](https://dequeuniversity.com/tips/provide-iframe-titles):

```
{{< youtube id="4Q7FTjhvZ7Y" title="Code Rush, a short documentary about the birth of the Mozilla project." >}}
```

**Review by schalkneethling — CHANGES_REQUESTED** [review]
Thank you, @bharmalh for this excellent write up. My apologies for having taken so long to get back to you. I have made some suggestions for you to consider.

File: content/topics/what-is-a-web-browser/index.md (hunk 2 above)
Comment (bharmalh) [reply content/topics/what-is-a-web-browser/index.md:13]: Just a small grammatical suggestion. I think the last word 'with' needs to be removed.

File: content/topics/what-is-a-web-browser/index.md (hunk 6 above)
Comment (bharmalh) [reply content/topics/what-is-a-web-browser/index.md:23]: But firefox extensions are not compatible with Chromium-based browsers. I wanted to mention it to show the relationship and benefits that all these browsers have because of their origins in Chromium.

File: content/topics/what-is-a-web-browser/index.md (hunk 2 above)
Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:13]: > Just a small grammatical suggestion. I think the last word 'with' needs to be removed.

Updated.

File: content/topics/what-is-a-web-browser/index.md (hunk 3 above)
Comment (schalkneethling) [reply content/topics/what-is-a-web-browser/index.md:15]: > I feel the article has more useful content besides just the video. I do agree with just linking to the page.

Ah, yes, I was not suggesting that we not link to the article. My suggestion was that we link to the article but also, embed the video on web://workbook. What do you think?

**Comment by schalkneethling** [comment]
Hey @bharmalh, thank you for all your work here. Let me know when you have had a moment to address the feedback. I would love to get this one merged. 🚀

**Comment by schalkneethling** [comment]
@bharmalh One thing to note, before making any changes locally, first do a pull from your branch, as there are some changes that were committed here and a merge of the latest from`main`.

**Comment by bharmalh** [comment]
Hi,
Sincere apologies for the delayed response. I have made the suggested changes and pushed. Kindly review.
Thanks.

**Review by schalkneethling — APPROVED** [review]
Thanks, @bharmalh 🎉
